pandas/tests/extension/base.py

```diff
@@ -0,0 +1,455 @@
+import operator
+
+import numpy as np
+import pytest
+
+import pandas as pd
+import pandas.util.testing as tm
+from pandas.compat import StringIO
+from pandas.core.internals import ExtensionBlock
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.core.dtypes.dtypes import ExtensionDtype
+
+
+class BaseDtypeTests(object):
```

Comment:
is there a reason to make these classes. I really hate this, we have been unwinding it across the codebase elsewhere. rather make sub-modules. its much much cleaner and easier to work with.

Reply:
further these for sure need splitting up, its way too long. make a sub-dir array and have test_indexing, test_construction, etc.

Reply:
> is there a reason to make these classes

So that we can ensure all our extension arrays (and third party ones) adhere to the interface by passing this test suite. I will split things up though.

Reply:
I think

Reply:
I would really really make another level here, its pretty simple

pandas/tests/extension/categorical/test_construction test_indexing.....etc
pandas/tests/extension/json
pandas/tests/extension/decimal

et.c

Reply:
I don't really understand your comment how you exactly propose to organize this. Can you explain in a bit more detail? 
If we are not using classes, it is much harder to let the different extension arrays run the same tests?

Reply:
we do this for indexes now. this is WAY too cumbersome to have tests like this all in one file. We have spent an enormous amount of time splitting them out. Please just split them out. You *can* have classes if you want. What I am after is short conscise test files that are completely obvious.

pandas/tests/extension/decimal/test_indexing.py
pandas/tests/extension/decimal/test_construction.py

etc.